psi4/src/psi4/mrcc/mrcc.cc

```diff
@@ -753,13 +758,17 @@ PsiReturnType mrcc_generate_input(SharedWavefunction ref_wfn, Options &options,
     spaces.push_back(MOSpace::all);
 
     // Check the reference.
-    bool restricted = true;
+    bool closedshell = ref_wfn->same_a_b_dens();
+    bool canonical = true;
 
-    if (options.get_str("REFERENCE") == "UHF")
-        restricted = false;
+    if (options.get_str("REFERENCE") == "ROHF")
+        canonical = false;
 
-    if (pertcc && options.get_str("REFERENCE") == "ROHF")
-        throw PSIEXCEPTION("Perturbative methods not implemented for ROHF references.");
+    if (pertcc && options.get_str("REFERENCE") == "ROHF") {
+        outfile->Printf("\n");
+        outfile->Printf("  WARNING: ROHF references are not implemented for perturbative\n");
+        outfile->Printf("           methods on older versions of MRCC. Proceed with caution.\n\n");
+    }
```

Comment:
Nothing that I know of, but in response to the new capabilities, as far as I can tell from the [2008 JCP paper,](http://aip.scitation.org/doi/10.1063/1.2988052) perturbative schemes for semicanonical references have been available for almost a decade.